pkg/steps/pod_test.go

```diff
@@ -182,3 +184,135 @@ func TestPodStepExecution(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPodObject(t *testing.T) {
+	testCases := []struct {
+		name        string
+		podStep     func(*podStep)
+		expectedPod func(*v1.Pod)
+	}{
+		{
+			name: "no secret",
```

Comment:
Give us cause and effect in test names please 

`"no secret name results in no mounted secrets"`